lib/wash/exe/wash.js

```diff
@@ -532,33 +533,53 @@ Wash.prototype.readEvalPrintLoop = function() {
 };
 
 /**
+ * @constructor
+ */
+Wash.RedirectInfo = function(operator, pathSpec) {
+  this.pathSpec = pathSpec;
+  this.mode = Wash.REDIR_OP_TO_MODE_[operator];
+  if (!this.pathSpec || !this.mode)
+    throw new AxiomError.Invalid('I/O redirection', operator + pathSpec);
+};
+
+/**
+ * @constructor
+ */
+Wash.CommandLine = function(pathSpec, argv, redirIn, redirOut) {
+  this.pathSpec = pathSpec;
+  this.argv = argv;
+  this.redirIn = redirIn;
+  this.redirOut = redirOut;
+};
+
+/**
+ * Matches a command line with possible stdout redirection to a file:
+ *
+ *    command [arg1 arg2 ...] [< some_file] [> some_file]
+ *
+ * @private
+ * @const
+ */
+Wash.CMD_WITH_IO_REDIRECT_RE_ = /^([^<>]+?)(?:\s+([^<>]+?))?\s*([<>].+)?$/;
```

Comment:
Yes, regexps :) If you feel uneasy about them, we can iterate, but right now I'm out of steam with this PR (in addition to not being a fan of the manual for-if-else-if-else-if kind of parsers).